### PR TITLE
Fix editorConfig for VB, Suppress IDE001 and IDE002, clean up VB Code and XML Document formatting issues

### PR DESCRIPTION
**Pull request by paul1956**
First Fix for #11215 VB Code Cleanup

## Proposed changes

- Suppress IDE001 and IDE002 in VB that will be needed to address Future PR feedback and make code easier to read
- Remove ByVal in VB Templates

## Customer Impact
They will not need to remove ByVal when creating new Forms from Templates

## Regression? 

- No

## Risk

- None

## Test methodology 
No code changes ByVal is VB Default and is not needed in source 

This simple PR will be followed by individual tests, that will test all changes in Future VB Changes

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11653)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -595,6 +595,12 @@ visual_basic_preferred_modifier_order = Partial,Default,Private,Protected,Public
 
 dotnet_separate_import_directive_groups = false:warning
 
+# IDE0001: Simplify Names
+dotnet_diagnostic.IDE0001.severity = silent
+
+# IDE0002: Simplify Member Access
+dotnet_diagnostic.IDE0002.severity = silent
```
Comment (lonitra) [inline .editorconfig:602]: Could you remind me why these needed to be suppressed?

Comment (paul1956) [reply .editorconfig:602]: Because in many places its clearer to have a more complex name. In Cleanup PR I have already simplified many of them that don't have suppressions and removed the suppress/restore Directives where they are not needed for clarity. For example I removed System.IO

```
Assert.Equal(System.IO.Path.GetDirectoryName(assembly.Location), assemblyInfo.DirectoryPath);
```
Assert.Equal(Path.GetDirectoryName(assembly.Location), assemblyInfo.DirectoryPath);
```
But I don't want to remove 175 places where the existing code is like 
```
 TypeOf ex Is IO.PathTooLongException OrElse
Return CType((UnsafeNativeMethods.GetKeyState(Keys.Scroll) And 1), Boolean)

```
and currently they are suppressed/restored manually in VB and the original author wants the IO before path or UnsafeNativeMethods to be explicate.

Comment (lonitra) [reply .editorconfig:602]: I see that makes total sense, thanks! It does seems like we want this analyzer on for some scenarios. Perhaps what we can do here is have a global using for the names that we do not want to simplify? In C#, we are able to define global usings like so: https://github.com/dotnet/winforms/blob/main/src/System.Drawing.Common/src/GlobalUsings.cs#L16
and use the name we gave it throughout the project. I'm not sure what the vb equivalent would look like (though I would assume similar) but perhaps we can do global usings for `IO` and `UnsafeNativeMethods` 🤔

Comment (paul1956) [reply .editorconfig:602]: In VB Global usings are per file below or per project like "System" (and I do use them because there are name conflicts sometimes) but this causes the same issue. VB wants to remove some of them.
```
Imports ExUtils = Microsoft.VisualBasic.CompilerServices.ExceptionUtils
```
**I want to do this change first but realized without fixing all the real violations it does not make sense. **

Comment (paul1956) [reply .editorconfig:602]: @lonitra If you look at #11215 that is the end result, I was trying to simplify PR into many small pieces so I could get it reviewed. But I believe that will cause a lot of extra work. It's really up to you, how you want to review.
#11215 has very few actual code changes just those absolutely required to support the tests and avoid conflict with Utils.

Comment (lonitra) [reply .editorconfig:602]: I am not the right person to be reviewing the VB tests + the network code, that would be @KlausLoeffelmann, but
I can definitely help out with reviewing the VB code clean up with all of the spelling issues, document comment clean up, etc. that you mentioned. @paul1956 would you be able to have a PR that contains only code clean up related changes? That way I can get that reviewed and merged. Etiher opening a new one or updating an existing one would work.

Comment (paul1956) [reply .editorconfig:602]: @lonitra if it's ok I will do all cleanup with this one. Is splitting 2 file into 4 covered or should that be separate? It was a comment from original Code Cleanup PR?

Comment (lonitra) [reply .editorconfig:602]: Of course, let's update this PR to be all clean up one then. If you are only splitting files, that is fine to include, it would be clear to have that as a separate commit for review.

Comment (paul1956) [reply .editorconfig:602]: Ok I have just committed the split files. I will do remainder of cleanup here.

Comment (paul1956) [reply .editorconfig:602]: @lonitra ready for review when you have time. It has the absolute minimum of code changes to make it still compile. There will be conflicts with the tests, that I will fix after merge. "Utils" need to be removed, I temporarily created an Alias VbUtils that will be changed when I add tests, it's just a place holder name.

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ApplicationBase.vb [hunk 2]
```diff
@@ -52,66 +53,69 @@ Namespace Microsoft.VisualBasic.ApplicationServices
...
-            Threading.Thread.CurrentThread.CurrentUICulture = New Globalization.CultureInfo(cultureName)
+            Thread.CurrentThread.CurrentUICulture = New Globalization.CultureInfo(cultureName)
         End Sub
 
-        Private _log As Logging.Log 'Lazy-initialized and cached log object.
-        Private _info As AssemblyInfo ' The executing application (the EntryAssembly)
+        'Lazy-initialized and cached log object.
+        Private _log As Logging.Log
+        ' The executing application (the EntryAssembly)
+        Private _info As AssemblyInfo
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ApplicationBase.vb:118]: Could these be moved to the beginning of the class?

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/CantStartSingleInstanceException.vb [hunk 3]
```diff
@@ -1,17 +1,14 @@
...
-Imports Microsoft.VisualBasic.CompilerServices.Utils
+
+Imports VbUtils = Microsoft.VisualBasic.CompilerServices.Utils
 
 Namespace Microsoft.VisualBasic.ApplicationServices
 
     ''' <summary>
-    ''' Exception for when we launch a single-instance application and it can't connect with the 
+    ''' Exception for when we launch a single-instance application and it can't connect with the
     ''' original instance.
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/CantStartSingleInstanceException.vb:12]:
```suggestion
    '''  Exception for when we launch a single-instance application and it can't connect with the
    '''  original instance.
```

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ConsoleApplicationBase.vb [hunk 4]
```diff
@@ -1,43 +1,47 @@
...
             MyBase.New()
         End Sub
 
         ''' <summary>
         '''  Returns the command line arguments for the current application.
         ''' </summary>
-        ''' <remarks>This function differs from System.Environment.GetCommandLineArgs in that the
-        ''' path of the executing file (the 0th entry) is omitted from the returned collection</remarks>
+        ''' <remarks>
+        '''  This function differs from System.Environment.GetCommandLineArgs in that the
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ConsoleApplicationBase.vb:33]:
```suggestion
        '''  This function differs from <see cref="System.Environment.GetCommandLineArgs"/> in that the
```

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ConsoleApplicationBase.vb [hunk 5]
```diff
@@ -47,8 +51,8 @@ Namespace Microsoft.VisualBasic.ApplicationServices
         End Property
 
         ''' <summary>
-        ''' Allows derived classes to set what the command line should look like.  WindowsFormsApplicationBase calls this
-        ''' for instance because we snag the command line from Main().
+        '''  Allows derived classes to set what the command line should look like. WindowsFormsApplicationBase calls this
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ConsoleApplicationBase.vb:54]:
```suggestion
        '''  Allows derived classes to set what the command line should look like. <see cref="WindowsFormsApplicationBase"/> calls this
```

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/User.vb [hunk 6]
```diff
@@ -30,7 +28,7 @@ Namespace Microsoft.VisualBasic.ApplicationServices
         End Property
 
         ''' <summary>
-        ''' The current IPrincipal which represents the current user
+        '''  The current IPrincipal which represents the current user
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/User.vb:31]:
```suggestion
        '''  The current <see cref="IPrincipal"/> which represents the current user.
```

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Helpers/NativeTypes.vb [hunk 7]
```diff
@@ -38,11 +35,12 @@ Namespace Microsoft.VisualBasic.CompilerServices
                 Dispose()
                 MyBase.Finalize()
             End Sub
+
         End Class
 
         ''' <summary>
-        ''' Inherits SafeHandleZeroOrMinusOneIsInvalid, with additional InitialSetHandle method.
-        ''' This is required because call to constructor of SafeHandle is not allowed in constrained region.
+        '''  Inherits SafeHandleZeroOrMinusOneIsInvalid, with additional InitialSetHandle method.
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Helpers/NativeTypes.vb:42]:
```suggestion
        '''  Inherits <see cref="SafeHandleZeroOrMinusOneIsInvalid"/>, with additional InitialSetHandle method.
```

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Interaction.vb [hunk 8]
```diff
@@ -237,127 +238,87 @@ Namespace Microsoft.VisualBasic
...
-                Dim FirstCommaLocation As Integer = FullName.IndexOf(","c)
-                If FirstCommaLocation >= 0 Then
-                    Title = FullName.Substring(0, FirstCommaLocation)
+                Dim firstCommaLocation As Integer = fullName.IndexOf(","c)
+                If firstCommaLocation >= 0 Then
+                    title = fullName.Substring(0, firstCommaLocation)
                 Else
                     'The name is not in the format we're expecting so return an empty string
-                    Title = ""
+                    title = ""
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Interaction.vb:297]:
```suggestion
                    title = string.Empty
```

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Logging/Log.vb [hunk 9]
```diff
@@ -233,8 +234,8 @@ Namespace Microsoft.VisualBasic.Logging
 
         ' Names of TraceSources
         Private Const WINAPP_SOURCE_NAME As String = "DefaultSource"
-        Private Const DEFAULT_FILE_LOG_TRACE_LISTENER_NAME As String = "FileLog" 'taken from appconfig
+        ' Taken from appConfig
+        Private Const DEFAULT_FILE_LOG_TRACE_LISTENER_NAME As String = "FileLog"
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Logging/Log.vb:238]: Let's move these and the above to the beginning of the class as well

File: .editorconfig [hunk 10]
```diff
@@ -595,6 +595,12 @@ visual_basic_preferred_modifier_order = Partial,Default,Private,Protected,Public
 
 dotnet_separate_import_directive_groups = false:warning
 
+# IDE0001: Simplify Names
+dotnet_diagnostic.IDE0001.severity = silent
```
Comment (lonitra) [inline .editorconfig:599]: I am still hesitant to silent these for all of our projects. Could we perhaps add a .editorconfig in the VisualBasic.Forms project and silence these just for that project?

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/AssemblyInfo.vb [hunk 11]
```diff
@@ -15,17 +13,36 @@ Namespace Microsoft.VisualBasic.ApplicationServices
     '''  specified using the assembly attributes (contained in AssemblyInfo.vb file in case of
     '''  a VB project in Visual Studio .NET).
     ''' </summary>
-    ''' <remarks>This class is based on the FileVersionInfo class of the framework, but
-    ''' reduced to a number of relevant properties.</remarks>
+    ''' <remarks>
+    '''  This class is based on the FileVersionInfo class of the framework, but
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/AssemblyInfo.vb:17]: Are we able to cref to FileVersionInfo?

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ApplicationBase.vb (hunk 2 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ApplicationBase.vb:118]: @lonitra I had planned a separate PR to move all privates to beginning of files (right now depending on the file they are all over the place and not even together) and sort them, but give the desire to have no code changes I thorough it would not be accepted in this PR. I use CodeMaid to do the sorting, some comment changes in this PR and to prepare for the Sort.

Please advise on how you want to handle the sorting, here or in a follow-on PR?

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Interaction.vb (hunk 8 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Interaction.vb:297]: @lonitra  I go back and forth on this, since .Net 2.0 they generate identical code in VB. I can do whatever you want.
Please advise and I will make them all the same?

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Logging/Log.vb (hunk 9 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Logging/Log.vb:238]: @lonitra based on this comment and others I will sort all the files using CodeMaid.

File: .editorconfig (hunk 10 above)
Comment (paul1956) [reply .editorconfig:599]: @lonitra It is only  in VB Projects
```

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ApplicationBase.vb (hunk 2 above)
Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ApplicationBase.vb:118]: Handling in a follow up PR sounds reasonable

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Interaction.vb (hunk 8 above)
Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Interaction.vb:297]: In c# we typically prefer string.Empty over "" imho we should prefer string.Empty here as well.

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ApplicationBase.vb (hunk 2 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/ApplicationBase.vb:118]: @lonitra thanks I think it is too big a change. I have done it in PR #9867 and it changes almost every line using automatic tool. All the new code is already properly sorted.

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/AssemblyInfo.vb (hunk 11 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/ApplicationServices/AssemblyInfo.vb:17]: Yes, I changed.

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Interaction.vb (hunk 8 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Interaction.vb:297]: OK I will do them all. No need to comment on each one, updating now.

**Comment by paul1956** [comment]
@lonitra by changing "" to String.Empty a lot of additional files have changed that I have already cleaned up in a follow-on PR

Do you want me to clean up those, now?

File: .editorconfig (hunk 10 above)
Comment (lonitra) [reply .editorconfig:599]: 😮 Ah I hadn't realized that it was VB only. In that case this change is fine with me.

**Comment by lonitra** [comment]
> @lonitra by changing "" to String.Empty a lot of additional files have changed that I have already cleaned up in a follow-on PR
> 
> Do you want me to clean up those, now?

Is the follow up PR code clean up only? If not, let's go ahead and take care of that in this PR to reduce overhead in the other.

File: src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/NetworkTests.cs [hunk 12]
```diff
@@ -13,10 +15,52 @@ public void IsAvailable()
...
+    {
+        Network network = new();
+        network.Ping(new Uri("http://127.0.0.1")).Should().BeTrue();
+    }
+
+    [Fact]
+    public void PingUri_Throw()
+    {
+        Network network = new();
+        Assert.Throws<ArgumentNullException>(() => network.Ping((Uri)null));
```
Comment (lonitra) [inline src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/NetworkTests.cs:56]: We can do fluent assertions for exceptions like so:
```suggestion
        Action action = () =>network.Ping((Uri)null);
        action.Should().Throw<ArgumentNullException>();
```

File: src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/NetworkTests.cs [hunk 13]
```diff
@@ -13,10 +15,52 @@ public void IsAvailable()
...
+    }
+
+    [Fact]
+    public void PingUriTimeout_Throw()
     {
         Network network = new();
-        Assert.True(network.Ping("127.0.0.1"));
+        Assert.Throws<ArgumentNullException>(() => network.Ping((Uri)null, 1));
```
Comment (lonitra) [inline src/Microsoft.VisualBasic/tests/UnitTests/Microsoft/VisualBasic/Devices/NetworkTests.cs:63]:
```suggestion
        Action action = () => network.Ping((Uri)null, 1));
        action.Should().Throw<ArgumentNullException>();
```

**Review by lonitra — APPROVED** [review]
LGTM 🥳 Please update the PR title to better reflect the changes here

**Comment by paul1956** [comment]
@lonitra  Do I update title on GitHub or with Visual Studio, still confused about process.

**Comment by lonitra** [comment]
Just on github. There should be an 'Edit' button next to the title of your PR at the top which will allow you to make changes to the PR title.

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/CompilerServices/Utils1.vb [hunk 14]
```diff
@@ -6,7 +6,7 @@ Imports System.Globalization
 Namespace Microsoft.VisualBasic.CompilerServices
 
     ' Purpose: various helpers for the vb runtime functions
-    Friend NotInheritable Class Utils
+    Friend NotInheritable Class Utils1
```
Comment (lonitra) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/CompilerServices/Utils1.vb:9]: What change occurred for this to now be clashing? It seemed like there was no clash previously.

Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/CompilerServices/Utils1.vb:9]: It clashes with the tests PR and just put it here to make merge easier latter. The entire class will go away, there are 3 functions here that conflict with another library and are only used locally and should be in ExceptionUtils. The Entire Exception Utils class is a mess (most of the code can't be reached any longer) and without all the tests I added it's easy to get the wrong Exception Thrown. I could do a separate PR to just replace ExceptionUtils and remove Utils but without all the tests it would be very dangerous.

Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/CompilerServices/Utils1.vb:9]: Below is the entire ExceptionUtils and Utils combined it is much simpler (25% less lines and 100% code coverage) and more in line with modern VB.
```
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.

Imports System.Threading

Namespace Microsoft.VisualBasic.CompilerServices

    Friend Enum vbErrors
        None = 0
        FileNotFound = 53
        PermissionDenied = 70
    End Enum

    ' Implements error utilities for Basic
    Friend Module ExceptionUtils

        Friend Function GetResourceString(resourceKey As String,
                            ParamArray args() As String) As String
            Return String.Format(Thread.CurrentThread.CurrentCulture, resourceKey, args)
        End Function

        Friend Function GetResourceString(resourceId As vbErrors) As String
            Dim id As String = $"ID{CStr(resourceId)}"
            Return SR.GetResourceString(id, id)
        End Function

        Friend Function VbMakeException(resourceId As Integer) As Exception
            Dim description As String = ""

            If resourceId > 0 AndAlso resourceId <= &HFFFFI Then
                description = GetResourceString(DirectCast(resourceId, vbErrors))
            End If

            Select Case resourceId

                Case vbErrors.FileNotFound
                    Return New IO.FileNotFoundException(description)

                Case vbErrors.PermissionDenied
                    Return New IO.IOException(description)

                Case Else
                    Return New Exception(description)
            End Select

        End Function

        ''' <summary>
        '''  Returns a new instance of <see cref="ArgumentException"/> with the message from resource file and the Exception.ArgumentName property set.
        ''' </summary>
        ''' <param name="argumentName">The name of the argument (parameter). Not localized.</param>
        ''' <param name="resourceID">The resource ID.</param>
        ''' <param name="placeHolders">Strings that will replace place holders in the resource string, if any.</param>
        ''' <returns>A new instance of <see cref="ArgumentException"/>.</returns>
        ''' <remarks>This is the preferred way to construct an argument exception.</remarks>
        Friend Function GetArgumentExceptionWithArgName(argumentName As String,
                            resourceID As String,
                            ParamArray placeHolders() As String) As ArgumentException

            Return New ArgumentException(GetResourceString(resourceID, placeHolders), argumentName)
        End Function

        ''' <summary>
        '''  Returns a new instance of <see cref="ArgumentNullException"/> with message: "Argument cannot be Nothing."
        ''' </summary>
        ''' <param name="argumentName">The name of the argument (parameter). Not localized.</param>
        ''' <returns>A new instance of <see cref="ArgumentNullException"/>.</returns>
        Friend Function GetArgumentNullException(argumentName As String) As ArgumentNullException

            Return New ArgumentNullException(argumentName, GetResourceString(SR.General_ArgumentNullException))
        End Function

        ''' <summary>
        '''  Returns a new instance of <see cref="ArgumentNullException"/> with the message from resource file.
        ''' </summary>
        ''' <param name="argumentName">The name of the argument (parameter). Not localized.</param>
        ''' <param name="resourceID">The resource ID.</param>
        ''' <param name="placeHolders">Strings that will replace place holders in the resource string, if any.</param>
        ''' <returns>A new instance of <see cref="ArgumentNullException"/>.</returns>
        Friend Function GetArgumentNullException(argumentName As String,
                            resourceID As String,
                            ParamArray placeHolders() As String) As ArgumentNullException

            Return New ArgumentNullException(argumentName, GetResourceString(resourceID, placeHolders))
        End Function

        ''' <summary>
        '''  Returns a new instance of <see cref="IO.DirectoryNotFoundException"/> with the message from resource file.
        ''' </summary>
        ''' <param name="resourceID">The resource ID.</param>
        ''' <param name="placeHolders">Strings that will replace place holders in the resource string, if any.</param>
        ''' <returns>A new instance of <see cref="IO.DirectoryNotFoundException"/>.</returns>
        Friend Function GetDirectoryNotFoundException(resourceID As String,
                            ParamArray placeHolders() As String) As IO.DirectoryNotFoundException

            Return New IO.DirectoryNotFoundException(GetResourceString(resourceID, placeHolders))
        End Function

        ''' <summary>
        '''  Returns a new instance of <see cref="IO.FileNotFoundException"/> with the message from resource file.
        ''' </summary>
        ''' <param name="fileName">The file name (path) of the not found file.</param>
        ''' <param name="resourceID">The resource ID.</param>
        ''' <param name="placeHolders">Strings that will replace place holders in the resource string, if any.</param>
        ''' <returns>A new instance of <see cref="IO.FileNotFoundException"/>.</returns>
        Friend Function GetFileNotFoundException(fileName As String,
                           resourceID As String,
                           ParamArray placeHolders() As String) As IO.FileNotFoundException

            Return New IO.FileNotFoundException(GetResourceString(resourceID, placeHolders), fileName)
        End Function

        ''' <summary>
        '''  Returns a new instance of <see cref="InvalidOperationException"/> with the message from resource file.
        ''' </summary>
        ''' <param name="resourceID">The resource ID.</param>
        ''' <param name="placeHolders">Strings that will replace place holders in the resource string, if any.</param>
        ''' <returns>A new instance of <see cref="InvalidOperationException"/>.</returns>
        Friend Function GetInvalidOperationException(resourceID As String,
                            ParamArray placeHolders() As String) As InvalidOperationException

            Return New InvalidOperationException(GetResourceString(resourceID, placeHolders))
        End Function

        ''' <summary>
        '''  Returns a new instance of <see cref="IO.IOException"/> with the message from resource file.
        ''' </summary>
        ''' <param name="resourceID">The resource ID.</param>
        ''' <param name="placeHolders">Strings that will replace place holders in the resource string, if any.</param>
        ''' <returns>A new instance of <see cref="IO.IOException"/>.</returns>
        Friend Function GetIOException(resourceID As String,
                            ParamArray placeHolders() As String) As IO.IOException

            Return New IO.IOException(GetResourceString(resourceID, placeHolders))
        End Function

        ''' <summary>
        '''  Returns a new instance of <see cref="ComponentModel.Win32Exception"/> with the message from resource file and the last Win32 error.
        ''' </summary>
        ''' <param name="resourceID">The resource ID.</param>
        ''' <param name="placeHolders">Strings that will replace place holders in the resource string, if any.</param>
        ''' <returns>A new instance of <see cref="ComponentModel.Win32Exception"/>.</returns>
        ''' <remarks>There is no way to exclude the Win32 error so this function will call Marshal.GetLastWin32Error all the time.</remarks>
        Friend Function GetWin32Exception(resourceID As String,
                            ParamArray placeHolders() As String) As ComponentModel.Win32Exception

            Return New ComponentModel.Win32Exception(Runtime.InteropServices.Marshal.GetLastWin32Error(), GetResourceString(resourceID, placeHolders))
        End Function

    End Module
End Namespace
```

Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/CompilerServices/Utils1.vb:9]: I see, would it be possible to have this change with the other PR? That way we are not splitting the reasoning behind the change into 2 PRs and it is centralized in one.

Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/CompilerServices/Utils1.vb]: Reverted change

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb [hunk 15]
```diff
@@ -584,15 +582,15 @@ Namespace Microsoft.VisualBasic.Devices
         ''' <param name="address">Uri representing the destination</param>
         ''' <param name="networkCredentials">The credentials of the user performing the upload</param>
         ''' <param name="showUI">Indicates whether or not to show a progress bar</param>
-        ''' <param name="connectionTimeout">Time alloted before giving up on a connection</param>
+        ''' <param name="connectionTimeout">Time allotted before giving up on a connection</param>
         ''' <param name="onUserCancel">Indicates what to do if user cancels dialog (either throw or do nothing)</param>
         Public Sub UploadFile(sourceFileName As String,
                               address As Uri,
```
Comment (KlausLoeffelmann) [inline src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:588]: Do we want to keep this kind of formatting or align it with C#?
(We can do that in a separate PR.)

**Review by KlausLoeffelmann — COMMENTED** [review]
This looks good to me in general, and I would not have any issues to take this.
I only looked here and there, but couldn't do a full review.
Doing this literately seems to be working this way, I actually like this!

@lonitra - if you feel OK with this, I am OK merging this!

**Comment by paul1956** [comment]
@KlausLoeffelmann I have a full set of tests that has close to 100% coverage that will be added after this is merged. All tests ran before the changes and also pass after changes.

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb (hunk 15 above)
Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:588]: We should align with C#, but let's do that in a separate PR as this is getting large.

**Review by lonitra — APPROVED** [review]
Thank you! 🥳

File: src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb (hunk 15 above)
Comment (paul1956) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:588]: Could someone explain how formatting differs? C# formatting, I thought is largely controlled by VS and .editorConfig, VB is manual but what are the differences that need to be addressed?

Comment (lonitra) [reply src/Microsoft.VisualBasic.Forms/src/Microsoft/VisualBasic/Devices/Network.vb:588]: When signatures are long and need to be broken up, our [C# style guidelines](https://github.com/dotnet/winforms/blob/main/docs/coding-style.md#line-breaks) is to break each argument of the signature on its own line with one indent in. Unfortunately we haven't gotten to specifying vb style guidelines as a team, but for the most part we'd want to align with C#. So, for this particular method we'd want to it to look like:
```vb
Public Sub UploadFile(
    sourceFileName As String,
    address As Uri,
    networkCredentials As ICredentials,
    showUI As Boolean,
    connectionTimeout As Integer,
    onUserCancel As UICancelOption)
```
